go/protocol/keybase1/extras.go

```diff
@@ -1774,6 +1786,61 @@ func (t TeamInviteType) String() (string, error) {
 	return "", nil
 }
 
+func TeamInviteTypeEq(a, b TeamInviteType) bool {
+	ac, err := a.C()
+	if err != nil {
+		return false
+	}
+	bc, err := b.C()
+	if err != nil {
+		return false
+	}
+	if ac != bc {
+		return false
+	}
+
+	switch ac {
+	case TeamInviteCategory_KEYBASE:
+		return true
+	case TeamInviteCategory_EMAIL:
+		return true
+	case TeamInviteCategory_SBS:
+		return a.Sbs() == b.Sbs()
+	case TeamInviteCategory_UNKNOWN:
+		return a.Unknown() == b.Unknown()
+	}
+
+	return false
+}
+
+func (t TeamInvite) KeybaseUserVersion() (UserVersion, error) {
```

Comment:
don't we already have something like this?

Reply:
yes, but it is in libkb for some reason.  I can move it.

Reply:
I think @mlsteele just moved it there!

Reply:
I moved `ParseUserVersion ` to libkb [here](https://github.com/keybase/client/blob/master/go/libkb/uid.go#L89). It has no dependencies on `teams` and I don't expect it ever will, so in libkb it can be used by other packages (which it isn't). I think that UserVersion is a Good Idea and I expect we will want it in more places as we add more systems.

Reply:
@mlsteele Any opposition to moving it to keybase1/extras.go?  Then even *more* places can use it.

Reply:
Anywhere you want :)